src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -476,32 +545,35 @@ private void PreProcessUserFunctionStatement(ParseNode node)
 
                 // build default dummy function to be used when this is a LOCK:
                 currentCodeSection = userFuncObject.GetUserFunctionOpcodes(0);
-                AddOpcode(new OpcodePush("$" + userFuncObject.Identifier)).Label = userFuncObject.DefaultLabel;
+                AddOpcode(new OpcodePush("$" + userFuncObject.ScopelessIdentifier)).Label = userFuncObject.DefaultLabel;
                 AddOpcode(new OpcodeReturn());
             }
 
             // lock expression's or function body's code
             currentCodeSection = userFuncObject.GetUserFunctionOpcodes(expressionHash);
-            if (isLock) // locks need to behave as if they had braces even though they don't - so they get lexical scope ids for closure reasons:
-                BeginScope(node);
-            if (isDefFunc)
-                nextBraceIsFunction = true;
+            bool secondInstanceSameLock = currentCodeSection.Count > 0;
```

Comment:
The first time the same exact lock expression is found, it fills the code with the expression contents.  The second time the same expression is used, it previously used to pointlessly append the same code again, that would never get reached because it would hit a "return" before encountering it.  This catches that case by seeing if the lock already got code compiled into it.  (the hashcode ensures that it's the same code body from the same scope level).  This works better than the old way, which assumed the lock would be unintialized the first time you encounter it.  That's not true anymore because we have to initialize the locks up front for them to be acessible from trigger bodies.